pkg/csi-handler/pv_handler_test.go

```diff
@@ -0,0 +1,283 @@
+		Controller: controllerServer,
+		Node:       nodeServer,
+	})
+	drv.StartOnAddress("unix", filepath.Join(tmpdir, "csi.sock"))
+
+	// Create a client connection to it
+	addr := drv.Address()
+	csiConn, err := New(addr)
+	if err != nil {
+		return nil, nil, nil, nil, nil, csiConnection{}, err
+	}
+
+	return mockController, drv, identityServer, controllerServer, nodeServer, csiConn, nil
+}
+
+func tempDir(t *testing.T) string {
+	dir, err := ioutil.TempDir("", "external-provisioner-test-")
+	if err != nil {
+		t.Fatalf("Cannot create temporary directory: %s", err)
+	}
+	return dir
+}
+
+func Test_csiPVHandler_ControllerListVolumeConditions(t *testing.T) {
+	tmpdir := tempDir(t)
+	defer os.RemoveAll(tmpdir)
+	mockController, driver, _, controllerServer, _, csiConn, err := createMockServer(t, tmpdir)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer mockController.Finish()
+	defer driver.Stop()
+
+	handler := NewCSIPVHandler(csiConn.conn)
+	in := &csi.ListVolumesRequest{
+		StartingToken: "",
+	}
+	out := &csi.ListVolumesResponse{
+		Entries: []*csi.ListVolumesResponse_Entry{
+			{
+				Volume: volume1,
+				Status: &csi.ListVolumesResponse_VolumeStatus{
+					VolumeCondition: abnormalVolumeCondition,
+				},
+			},
+			{
+				Volume: volume2,
+				Status: &csi.ListVolumesResponse_VolumeStatus{
+					VolumeCondition: normalVolumeCondition,
+				},
+			},
+		},
+		NextToken: "",
+	}
+
+	controllerServer.EXPECT().ListVolumes(gomock.Any(), in).Return(out, nil).Times(1)
+	tests := []struct {
+		name    string
+		want    map[string]*VolumeConditionResult
+		wantErr bool
+	}{
+		{
+			name: "case1",
+			want: map[string]*VolumeConditionResult{
+				"1": {
+					abnormal: true,
+					message:  "The volume not found",
+				},
+				"2": {
+					abnormal: false,
+					message:  "",
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := handler.ControllerListVolumeConditions(context.Background())
+			if (err != nil) != tt.wantErr {
+				t.Errorf("csiPVHandler.ControllerListVolumeConditions() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("csiPVHandler.ControllerListVolumeConditions() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_csiPVHandler_ControllerGetVolumeCondition(t *testing.T) {
+	tmpdir := tempDir(t)
+	defer os.RemoveAll(tmpdir)
+	mockController, driver, _, controllerServer, _, csiConn, err := createMockServer(t, tmpdir)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer mockController.Finish()
+	defer driver.Stop()
+
+	handler := NewCSIPVHandler(csiConn.conn)
+	in := &csi.ControllerGetVolumeRequest{
+		VolumeId: "1",
+	}
+	out := &csi.ControllerGetVolumeResponse{
+		Volume: volumeMap["1"].Volume,
+		Status: &csi.ControllerGetVolumeResponse_VolumeStatus{
+			VolumeCondition: volumeMap["1"].Condition,
+		},
+	}
+	controllerServer.EXPECT().ControllerGetVolume(gomock.Any(), in).Return(out, nil).Times(1)
```

Comment:
Need csi-test PR merged

Reply:
The csi-test PR https://github.com/kubernetes-csi/csi-test/pull/276 is merged.

Reply:
Great!